lib/midcom/helper/imagepopup/handler/upload.php

```diff
@@ -0,0 +1,181 @@
+<?php
+use Symfony\Component\HttpFoundation\File\File;
+
+/**
+ * @author lukasz chalat
+ * @package midcom.helper.imagepopup
+ * @copyright The Midgard Project, http://www.midgard-project.org
+ * @license http://www.gnu.org/licenses/lgpl.html GNU Lesser General Public License
+ */
+
+/**
+ * This handler get uploaded file and save it in database and write to file.
+ *
+ * @package midcom.helper.imagepopup
+ */
+class midcom_helper_imagepopup_handler_upload extends midcom_baseclasses_components_handler
+{
+    /**
+     *
+     * @param mixed $handler_id The ID of the handler.
+     * @param array $args The argument list.
+     * @param array &$data The local request data.
+     */
+    public function _handler_upload($handler_id, array $args, array &$data)
+    {
+        // Get the file
+        reset($_FILES);
+        $temp = array_shift($_FILES);
+        
+        $accepted_origin = midcom::get()->get_host_name();
+        
+        if(is_uploaded_file($temp['tmp_name']))
+        {
+            if(isset($_SERVER['HTTP_ORIGIN']))
+            {
+                // same-origin requests won't set an origin. If the origin is set, it must be valid.
+                if(in_array($_SERVER['HTTP_ORIGIN'], $accepted_origin))
```

Comment:
also, per your comment "same-origin requests won't set an origin". so isn't this whole block redundant? i.e. if we come from the same origin, the header won't be sent, so we never get into this check. if we come from any other origin, it will throw an exception